app.rb

```diff
@@ -0,0 +1,167 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @class = Classroom.new('Grade 9')
+  end
+
+  def run
+    puts 'Welcome to Blessed jason Mwanza School Library!'
+    loop do
+      menu
+      option = gets.chomp
+      break if option == '7'
+
+      get_num option
+    end
+    puts 'Thank you for using our Library!'
+  end
+
+  def menu
+    puts
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    puts 'There are no books yet! Kindly add books.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    puts 'There are no people yet! Kindly add a student or teacher.' if @people.empty?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Kindly type 1 or 2'
+    end
+  end
```

Comment:
- [ ] When I try to create a new person, it threw an error after I entered the parent permission! Please double-check the `person.rb` file!
![Git-Bksh__c_Users_mbakr_OneDrive_Desktop_assesss_school-library-ruby 3_18_2022 12_51_54 AM (1)](https://user-images.githubusercontent.com/47625407/158907438-ee78a009-0084-4100-a0dc-d2fb619d305d.png)

Reply:
Thanks a lot for this point out @MahmoudBakr23 . I have fixed the issue and now you can create a persons account

Reply:
This issue is well resolved! ✔️